weld-python/weld/lazy.py

```diff
@@ -301,3 +302,83 @@ def identity(phys_value, decoder):
     Creates an identity `WeldLazy` from a `PhysicalValue` and decoder.
     """
     return WeldLazy(str(phys_value.id), [phys_value], phys_value.output_type, decoder)
+
+class WeldFunc(object):
+    """
+    The result of a function annotated with `welfunc`.
+
+    This object is effectively a function with a field `code`, which extracts the raw code
+    constructed by the annotated function without constructing a `WeldLazy`.
+
+    """
+
+    __slots__ = ["_code", "_dependencies"]
```

Comment:
Why do you need this?

Reply:
This represents the object as a tuple instead of a dict, so its slightly more efficient to construct the object (it also prevents adding other attributes unless the `__slots__` definition is updated). Using it here since we don't want to make the creation of this wrapper object expensive.

Reply:
Interesting